Remove the revCount attribute from the generated flake registry

r-vdp opened the pull request:
## Description of changes

The revCount attribute is only present when the flake is an actual git
repo and not when using a flake reference with an input scheme for a git
forge that downloads tarballs instead (like github: or sourcehut:).
Because of this, the hash of the nixos config changes depending on
whether the config was built from a local git clone or from a flake ref
using one of those input schemes.
This leads to confusion (I lost quite some time figuring out what
happened here) and unnecessary rebuilds.
To my knowledge, there is no immediate benefit to including revCount in
the flake registry entry.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-vdp committed:
Remove the revCount attribute from the generated flake registry

The revCount attribute is only present when the flake is an actual git
repo and not when using a flake reference with an input scheme for a git
forge that downloads tarballs instead (like github: or sourcehut:).
Because of this, the hash of the nixos config changes depending on
whether the config was built from a local git clone or from a flake ref
using one of those input schemes.
This leads to confusion (I lost quite some time figuring out what
happened here) and unnecessary rebuilds.
To my knowledge, there is no immediate benefit to including revCount in
the flake registry entry.

lf- reviewed (commented):
wow yeah. so i think that there is actually revCount from the forge ones, in principle that's supposed to be the case at least, i think. but i also agree this is utterly useless in a flake registry entry on disk. it is honestly news to me that this is winding up in the json file at all though?

I'm going to go look at this a bit harder but I generally like this change.

r-vdp commented:
Well at least for sourcehut, I don't get a `revCount` when I build using a `sourcehut:` reference, but I do get one when I build from a local clone:
```
Lix 2.90.0-beta.1
Type :? for help.
nix-repl> :lf sourcehut:~r-vdp/nixos-config
Added 21 variables.

nix-repl> sourceInfo
{
  lastModified = 1717232731;
  lastModifiedDate = "20240601090531";
  narHash = "sha256-zCzXuwkyGoQDRvCSkOi2uv1NMd+yZjufZG6lzGVP5AA=";
  outPath = "/nix/store/9ilkiqfx5m85riwmnc6b0p4qdrwy4s95-source";
  rev = "23b24e1e7d993cb5ec005a3a22345524f9579c39";
  shortRev = "23b24e1";
}

nix-repl> :lf .
Added 23 variables.

nix-repl> sourceInfo
{
  lastModified = 1717232731;
  lastModifiedDate = "20240601090531";
  narHash = "sha256-zCzXuwkyGoQDRvCSkOi2uv1NMd+yZjufZG6lzGVP5AA=";
  outPath = "/nix/store/9ilkiqfx5m85riwmnc6b0p4qdrwy4s95-source";
  rev = "23b24e1e7d993cb5ec005a3a22345524f9579c39";
  revCount = 2319;
  shortRev = "23b24e1";
  submodules = false;
}
```

r-vdp commented:
@lf- have you had the time to give this some more thought?
